deepmd/dpmodel/utils/safe_gradient.py

```
@@ -0,0 +1,32 @@
+    mask = x > 0.0
+    return xp.where(mask, xp.sqrt(xp.where(mask, x, xp.ones_like(x))), xp.zeros_like(x))
+
+
+def safe_for_vector_norm(x, /, *, axis=None, keepdims=False, ord=2):
+    """Safe version of sqrt that has a gradient of 0 at x = 0."""
+    xp = array_api_compat.array_namespace(x)
+    mask = xp.sum(xp.square(x), axis=axis, keepdims=True) > 0
+    if keepdims:
+        mask_squeezed = mask
+    else:
+        mask_squeezed = xp.squeeze(mask, axis=axis)
```

Ruff

Use ternary operator `mask_squeezed = mask if keepdims else xp.squeeze(mask, axis=axis)` instead of `if`-`else`-block

Replace `if`-`else`-block with `mask_squeezed = mask if keepdims else xp.squeeze(mask, axis=axis)`

(SIM108)